source/margarita/states.rst

```diff
@@ -0,0 +1,621 @@
+
+Always adds the elasticsearch-analysis-icu plugin.
+
+Pillar configuration:
+
+* ``elasticsearch_version`` (string):  E.g. '1.7'
+* ``elasticsearch_newrelic`` (boolean): Whether to use a newrelic plugin to monitor
+  the elasticsearch service in more detail.
+
+Dependencies:
+
+* :ref:`newrelic_npi` (if ``elasticsearch_newrelic`` is True).
+
+.. _fail2ban:
+
+fail2ban
+~~~~~~~~
+
+Installs and runs fail2ban.
+
+No configuration.
+
+.. _forward_logs:
+
+forward_logs
+~~~~~~~~~~~~
+
+Arranges for all syslog messages to be copied to a remote server
+(e.g. Papertrail).
+
+Pillar configuration:
+
+* ``secrets:LOG_DESTINATION`` (string): The host:port to forward the messages to,
+  e.g. "log3.example.com:12345".
+
+.. _locale.utf8:
+
+locale.utf8
+~~~~~~~~~~~
+
+Sets the system locale, both ``LANG`` and ``LC_ALL``, to ``en_US.UTF-8``.
+
+.. _memcached:
+
+memcached
+~~~~~~~~~
+
+Installs and runs memcached.
+
+Leaves all settings and configuration at their defaults.
+
+.. _newrelic_npi:
+
+newrelic_npi
+~~~~~~~~~~~~
+
+Install the `New Relic Platform Installer <https://docs.newrelic.com/docs/plugins/plugins-new-relic/installing-plugins/installing-npi-compatible-plugin>`_
+that can be used to easily install compatible plugins.
+
+This can be pre-reqed by other states but probably does not need to
+be included directly.
+
+.. _newrelic_sysmon:
+
+newrelic_sysmon
+~~~~~~~~~~~~~~~
+
+Install and configure the NewRelic monitor for the whole system
+(monitors CPU, memory usage, etc.)
+
+Note that if the key is not set in pillar, this state is a no-op, so
+it's safe to always include it and then just set the key on the
+systems where you want to monitor.
+
+Pillar configuration:
+
+* ``secrets:NEW_RELIC_LICENSE_KEY`` (string): The license key of the New Relic
+  account to use.
+
+.. _nginx:
+
+nginx
+~~~~~
+
+Install nginx, tweak the default configuration in the main config file,
+and remove the default site config that gets installed.
+
+Uses the Nginx PPA to get the latest stable version.
+
+.. _nginx.cert:
+
+nginx.cert
+~~~~~~~~~~
+
+Installs (but does not run) a script, ``/var/lib/nginx/generate-cert.sh``,
+for creating self-signed certificates for nginx.
+
+Dependencies:
+
+* :ref:`nginx`
+
+.. _nodejs:
+
+nodejs
+~~~~~~
+
+Installs node, making sure there's a ``/usr/bin/node`` executable.
+
+.. _postfix:
+
+postfix
+~~~~~~~
+
+Install and run postfix mail system.
+
+.. _postgresql:
+
+postgresql
+~~~~~~~~~~
+
+Install the configured version of the Postgres database server.  For older
+versions, makes sure that UTF-8 locale is the default for the main cluster.
+(For newer versions, it comes set up that way.)
+
+Pillar configuration:
+
+* ``postgres_version`` (string): Postgres version to install, e.g. "9.4".
+
+Dependencies:
+
+* :ref:`locale.utf8`
+
+.. _project.cache:
+
+project.cache
+~~~~~~~~~~~~~
+
+Open the firewall for memcached from :ref:`app_minions <minions>`.
+
+Dependency on `memcached` state will also ensure memcached is installed.
```

Comment:
Actually my RST knowledge is fuzzy... Maybe that was supposed to be a "link" to the memcached state?